zebra/zebra_nhg.c

```diff
@@ -2210,24 +2257,23 @@ static int nexthop_active(struct nexthop *nexthop, struct nhg_hash_entry *nhe,
 		}
 
 		if (match->type == ZEBRA_ROUTE_CONNECT) {
-			/* Directly point connected route. */
-			newhop = match->nhe->nhg.nexthop;
-			if (newhop) {
-				if (nexthop->type == NEXTHOP_TYPE_IPV4
-				    || nexthop->type == NEXTHOP_TYPE_IPV6)
-					nexthop->ifindex = newhop->ifindex;
-				else if (nexthop->ifindex != newhop->ifindex) {
-					if (IS_ZEBRA_DEBUG_RIB_DETAILED)
-						zlog_debug(
-							"%s: %pNHv given ifindex does not match nexthops ifindex found found: %pNHv",
-							__func__, nexthop,
-							newhop);
-					/*
-					 * NEXTHOP_TYPE_*_IFINDEX but ifindex
-					 * doesn't match what we found.
-					 */
-					return 0;
-				}
+			newhop = zebra_nhg_connected_ifindex(rn, match,
```

Comment:
So this could result in a situation where 'newhop' is not part of the 'match' route_entry, right? is there anyplace below where that could become a problem, where there's an assumption that 'newhop' is part of 'match' ?

Reply:
newhop is only used in this section.

Reply:
'match' and its nhg id are used in the debug at the end of this clause, for example? I was thinking that maybe it's 'match' that should be returned by the new utility api - that way you know you've got the actual route that you think will be suitable?

Reply:
but by it's very definition it's the same route?

Reply:
this whole section is not ideal.  I'll see what I can do